### ISSUE TITLE
Availability Service timewindow shifted by timezone

### ISSUE THREAD

**Issue by jvh003** [OfficeDev/ews-java-api#241]
The `getUserAvailability()` api is not returning the expected results/events.
It seems that the timewindow argument is mis-interrepted as UTC instead of my local timezone.

For this test I have 4 meeting events configured for user ps5502 (using EST) as follows:

Event1: March 4th 19:00 to 20:00
Event2: March 5th 10:00 to 11:00
Event3: March 5th 14:00 to 17:00
Event4: March 5th 19:00 to 19:30

I'm testing against a 2010 exchange server and you can see from this test code output
that I'm getting back **events 1, 2, 3** where I should be getting **events 2, 3, 4**
for the specified time window (which is from the start of today to the start of tomorrow - 24hrs).

```
2015-03-05 18:50:09,490 [main] jvh.ewsClient.EWSClient INFO - buildStartEndTime() start Thu Mar 05 00:00:00 EST 2015 end Fri Mar 06 00:00:00 EST 2015
2015-03-05 18:50:10,099 [main] jvh.ewsClient.EWSClient INFO - getAvailability() start Thu Mar 05 00:00:00 EST 2015 end Fri Mar 06 00:00:00 EST 2015
2015-03-05 18:50:10,099 [main] jvh.ewsClient.EWSClient INFO - getAvailability() results for ps5502@ca.avaya.com events: 3
2015-03-05 18:50:10,099 [main] jvh.ewsClient.EWSClient INFO - getAvailability()    event 1 start: Wed Mar 04 19:00:00 EST 2015 end: Wed Mar 04 20:00:00 EST 2015
2015-03-05 18:50:10,100 [main] jvh.ewsClient.EWSClient INFO - getAvailability()    event 2 start: Thu Mar 05 10:00:00 EST 2015 end: Thu Mar 05 11:00:00 EST 2015
2015-03-05 18:50:10,100 [main] jvh.ewsClient.EWSClient INFO - getAvailability()    event 3 start: Thu Mar 05 14:00:00 EST 2015 end: Thu Mar 05 17:00:00 EST 2015
```

Test code as follows:

    private TimeWindow buildStartEndTime()
    {
       final Calendar cal = Calendar.getInstance();
       cal.set(Calendar.HOUR_OF_DAY, cal.getActualMinimum(Calendar.HOUR_OF_DAY));
       cal.set(Calendar.MINUTE, cal.getActualMinimum(Calendar.MINUTE));
       cal.set(Calendar.SECOND, cal.getActualMinimum(Calendar.SECOND));
       cal.set(Calendar.MILLISECOND, cal.getActualMinimum(Calendar.MILLISECOND));
       final Date start = cal.getTime();
       cal.add(Calendar.DATE, 1);
       final Date end = cal.getTime();
 
       LOG.info("buildStartEndTime() start " + start + " end " + end);
       return new TimeWindow(start, end);
    }
    
    private void testAvailabilityService()
    {
       final List<String> attendeeList = Arrays.asList("ps5502@ca.avaya.com");
       final TimeWindow tw = buildStartEndTime();
       getAvailability(attendeeList, tw);
    }
    
    private void getAvailability(List<String> attendeeList, final TimeWindow tw)
    {
       final List<AttendeeInfo> attendees = new ArrayList<AttendeeInfo>();
       for (final String attendee : attendeeList)
       {
          attendees.add(new AttendeeInfo(attendee));
       }
 
       try
       {
          GetUserAvailabilityResults results = service.getUserAvailability(attendees, tw, AvailabilityData.FreeBusyAndSuggestions);
 
          LOG.info("getAvailability() start " + tw.getStartTime() + " end " + tw.getEndTime());
          int userIndex = 0;
          for (AttendeeAvailability attendeeAvailability : results.getAttendeesAvailability())
          {
             if (attendeeAvailability.getErrorCode() == ServiceError.NoError)
             {
                LOG.info("getAvailability() results for " + attendees.get(userIndex).getSmtpAddress() + " events: " + attendeeAvailability.getCalendarEvents().size());
 
                // loop thru the rec'd events
                int eventCount = 1;
                for (CalendarEvent calEvent : attendeeAvailability.getCalendarEvents())
                {
                   LOG.info("getAvailability()    event " + eventCount + " start: " + calEvent.getStartTime() + " end: " + calEvent.getEndTime());  // + " " + " details: " + calEvent.getDetails() + " subject: " + (calEvent.getDetails() == null ? "null" : calEvent.getDetails().getSubject()));
                   eventCount++;
                }
             }
             else
             {
                LOG.info("getAvailability() error for " + attendees.get(userIndex).getSmtpAddress() + ": "+ attendeeAvailability.getErrorCode());
             }
             userIndex++;
          }
       }
       catch (Exception e)
       {
          LOG.error("getAvailability() failed", e);
       }
    }

BTW... I executed this same test with the older version 1.1.5 and it does the same thing.
However, the old version sends back calendar events in UTC where the lastest 2.0 code
sends the events back in my local timezone (ie: EST).  (which is better :-) )

Test results using the 1.1.5 version:

```
2015-03-05 19:16:07,743 [main] jvh.ewsClient.EWSClient INFO - buildStartEndTime() start Thu Mar 05 00:00:00 EST 2015 end Fri Mar 06 00:00:00 EST 2015
2015-03-05 19:16:09,568 [main] jvh.ewsClient.EWSClient INFO - getAvailability() start Thu Mar 05 00:00:00 EST 2015 end Fri Mar 06 00:00:00 EST 2015
2015-03-05 19:16:09,568 [main] jvh.ewsClient.EWSClient INFO - getAvailability() results for ps5502@ca.avaya.com events: 3
2015-03-05 19:16:09,568 [main] jvh.ewsClient.EWSClient INFO - getAvailability()    event 1 start: Thu Mar 05 00:00:00 EST 2015 end: Thu Mar 05 01:00:00 EST 2015
2015-03-05 19:16:09,569 [main] jvh.ewsClient.EWSClient INFO - getAvailability()    event 2 start: Thu Mar 05 15:00:00 EST 2015 end: Thu Mar 05 16:00:00 EST 2015
2015-03-05 19:16:09,570 [main] jvh.ewsClient.EWSClient INFO - getAvailability()    event 3 start: Thu Mar 05 19:00:00 EST 2015 end: Thu Mar 05 22:00:00 EST 2015
```

**Comment by jvh003** [comment]
Did a little more testing here... and found that's not a simple shifting of TZ.
I expanded my test code to try 2 different things:
1) a sliding time window - which didn't help.
2) an expanded time window - which eventually did work, but picks up calendar events that are really outside the real window I'm interested in (which is one complete day's worth of events).

For example:
In order to get all the events for 1 single day, I had time window that is actually 2 days wide.
Here's the output from my test code (with time window from March26 to March 28):
```
2015-03-26 15:24:40,140 [main] jvh.ewsClient.EWSClient INFO - getAvailability() start Thu Mar 26 00:00:00 EDT 2015 end Sat Mar 28 00:00:00 EDT 2015
2015-03-26 15:24:40,140 [main] jvh.ewsClient.EWSClient INFO - getAvailability() results for ps5502@ca.avaya.com events: 6
2015-03-26 15:24:40,140 [main] jvh.ewsClient.EWSClient INFO - getAvailability()    event 1 start: Wed Mar 25 19:00:00 EDT 2015 end: Wed Mar 25 20:00:00 EDT 2015 status: OOF
2015-03-26 15:24:40,140 [main] jvh.ewsClient.EWSClient INFO - getAvailability()    event 2 start: Thu Mar 26 10:00:00 EDT 2015 end: Thu Mar 26 11:00:00 EDT 2015 status: Tentative
2015-03-26 15:24:40,140 [main] jvh.ewsClient.EWSClient INFO - getAvailability()    event 3 start: Thu Mar 26 14:00:00 EDT 2015 end: Thu Mar 26 17:00:00 EDT 2015 status: Free
2015-03-26 15:24:40,140 [main] jvh.ewsClient.EWSClient INFO - getAvailability()    event 4 start: Thu Mar 26 19:00:00 EDT 2015 end: Thu Mar 26 19:30:00 EDT 2015 status: Busy
2015-03-26 15:24:40,140 [main] jvh.ewsClient.EWSClient INFO - getAvailability()    event 5 start: Thu Mar 26 23:00:00 EDT 2015 end: Fri Mar 27 00:00:00 EDT 2015 status: Busy
2015-03-26 15:24:40,140 [main] jvh.ewsClient.EWSClient INFO - getAvailability()    event 6 start: Fri Mar 27 17:00:00 EDT 2015 end: Fri Mar 27 18:00:00 EDT 2015 status: Busy
```

From this test, I really am only interested in events 2,3,4, & 5 (all on March 26th).
But I get event 1 (from the previous day 25th) and in order to get event 5 (by expanding the time window) I also get event 6 (for the next day 27th).

**Comment by serious6** [comment]
Hi @jvh003 did you also consider the request / response? This might indicate if there is a bug on the api side or if this might be a bug with exchange. Perhabs you also can use the [ews editor](https://ewseditor.codeplex.com) to underline this.

**Comment by jvh003** [comment]
Thanks @serious6, that was a good suggestion. (Had not used EWSEditor before).
So executing the same test produces the correct results. I get the calendar events from the server that I expect in the window that I had asked for.
If supplying the ews-request & ews-response traces would help, please let me know.

**Comment by cfraser** [comment]
@jvh003 I think this issue may be resolved by #396. I'll add your test case to my test this weekend sometime to verify.

**Comment by cfraser** [comment]
Looking at the code it appears that the TimeWindow is GetUserAvailability only supports full days. This fix should resolve the time window correctly although it won't behave exactly as you expected.

There's also a comment [here](https://social.msdn.microsoft.com/Forums/exchange/en-US/27701d88-3f4e-48ad-9364-f6006a777f34/bug-in-ews-managed-api-getuseravailability-soap-call-does-not-include-time-component-of?forum=exchangesvrdevelopment) from the Program Manager of Exchange Web Services David Claux that corroborates that it only supports 24 hour blocks.

I hope that helps.
